AmazonChimeSDK/AmazonChimeSDK/ingestion/IngestionEventConverter.swift

```diff
@@ -0,0 +1,165 @@
+                                                                               poorConnectionCount: eventAttributes[EventAttributeName.poorConnectionCount] as? Int,
+                                                                               retryCount: eventAttributes[EventAttributeName.retryCount] as? Int,
+                                                                               videoInputError: videoErrorStr,
+                                                                               meetingId: meetingConfig?.meetingId,
+                                                                               attendeeId: meetingConfig?.attendeeId))
+    }
+
+    func toIngestionRecord(dirtyMeetingEvents: [DirtyMeetingEventItem]) -> IngestionRecord {
+        if dirtyMeetingEvents.isEmpty {
+            return IngestionRecord(metadata: IngestionMetadata(), events: [])
+        }
+
+        let dirtyMeetingEventsGrouped = Dictionary(grouping: dirtyMeetingEvents, by: { $0.data.eventAttributes.meetingId })
+
+        let ingestionEvents = dirtyMeetingEventsGrouped.compactMap { (group) -> IngestionEvent? in
+            if let dirtyMeetingEventItem = group.value.first {
+                return IngestionEvent(type: String(describing: EventClientType.meet),
+                               metadata: toIngestionMetadata(ingestionMeetingEvent: dirtyMeetingEventItem.data),
+                               payloads: group.value.map { dirtyMeetingEvent in
+                                   toIngestionPayload(dirtyMeetingEvent: dirtyMeetingEvent)
+                               })
+            }
+            return nil
+        }
+
+        let rootMeta = toIngestionMetadata(attributes: EventAttributeUtils.getCommonAttributes())
+
+        return IngestionRecord(metadata: rootMeta,
+                               events: ingestionEvents)
+    }
+
+    func toIngestionRecord(meetingEvents: [MeetingEventItem]) -> IngestionRecord {
+        if meetingEvents.isEmpty {
+            return IngestionRecord(metadata: IngestionMetadata(), events: [])
+        }
+        let meetingEventsByMeetingId = Dictionary(grouping: meetingEvents, by: { $0.data.eventAttributes.meetingId })
+
+        let type = String(describing: EventClientType.meet)
+
+        // When sending a batch, server accepts the data as
+        // "events": [
+        //   {
+        //      "metadata": { "meetingId": "meetingId2" } // This overrides record level metadata
+        //      "type": "Meet",
+        //      "v": 1,
+        //      "payloads": []
+        //   },
+        //   {
+        //      "metadata": { "meetingId": "meetingId1" } // This overrides record level metadata
+        //      "type": "Meet",
+        //      "v": 1,
+        //      "payloads": []
+        //   },
+        // ]
+        // This is to group events so that it contains different metadata to override
```

Comment:
comment still needed ?

Reply:
Yes since it gives overview how format will look like and why we group